src/DependencyInjection/Compiler/VarnishCachePass.php

```diff
@@ -38,4 +41,15 @@ private function processVarnishProxyClientSettings(ContainerBuilder $container)
             $baseUrl
         );
     }
+
+    private function overrideDefaultProxyClient(ContainerBuilder $container): void
+    {
+        $varnishProxyClient = $container->getDefinition('fos_http_cache.proxy_client.varnish');
+        $varnishProxyClient->setClass(Varnish::class);
+        $varnishProxyClient->setArguments([
+            new Reference('ezpublish.config.resolver'),
+            new Reference('fos_http_cache.proxy_client.varnish.http_dispatcher'),
+            $container->getParameter('fos_http_cache.proxy_client.varnish.options'),
+        ]);
```

Comment:
Could you decorate (via `decorates` attribute in Yaml) instead of overriding?

Reply:
I tried, but that class is very convoluted with Abstract.

Reply:
Ok, let's not spend too much time on this now. Thanks for trying :+1: